### DISCUSSION TITLE
Support to older DC

### DISCUSSION THREAD

**Discussion by JuniorDark** [vezel-dev/novadrop#17] [Questions]
`Unhandled exception: System.IO.InvalidDataException: The archive entry was compressed using an unsupported compression method.`

novadrop-dc is working great on version 105, would be possible to add support for older versions like 100. Thanks.

**Comment by alexrp** [comment]
Out of curiosity, what's the use case for opening older data centers now that TERA is shutting down? AFAIK, the developers rarely outright delete data from them, so it's not obvious to me why you would actually need to open old data centers.

In any case, I *think* data center files from patch 100 have the same format as patch 105. I just need to add some options to novadrop-dc that let you pass in the encryption key/IV; right now, it always uses the key/IV from patch 105, which is why you get that `InvalidDataException` when trying to decompress it (the decrypted data is nonsense). (See #18.)

**Comment by justkeepquiet** [comment]
Maybe patch 100.02 will be used on private servers after the recent leak of the retail version. DC repacking is needed to be able to translate StrSheets into other languages. The format of 100 the same as 105. Yes, you need to add the parameters to specify key/iv.

**Comment by justkeepquiet** [comment]
iv/key for 100.02:

src\formats\Data\DataCenter.cs
```cs
    public static ReadOnlyMemory<byte> LatestKey { get; } = new byte[]
    {
        0x1c, 0x01, 0xc9, 0x04, 0xff, 0x76, 0xff, 0x06,
        0xc2, 0x11, 0x18, 0x7e, 0x19, 0x7b, 0x57, 0x16,
    };

    public static ReadOnlyMemory<byte> LatestIV { get; } = new byte[]
    {
        0x39, 0x6c, 0x34, 0x2c, 0x52, 0xa0, 0xc1, 0x2d,
        0x51, 0x1d, 0xd0, 0x20, 0x9f, 0x90, 0xca, 0x7d,
    };
```
Updated XSDs for 100.02: https://drive.google.com/drive/folders/1U1w4GgEgEoeayX3dmMryIX70ztdisCl9?usp=sharing

**Reply by JuniorDark** [reply]
> iv/key for 100.02:
> 
> src\formats\Data\DataCenter.cs
> 
> ```cs
>     public static ReadOnlyMemory<byte> LatestKey { get; } = new byte[]
>     {
>         0x1c, 0x01, 0xc9, 0x04, 0xff, 0x76, 0xff, 0x06,
>         0xc2, 0x11, 0x18, 0x7e, 0x19, 0x7b, 0x57, 0x16,
>     };
> 
>     public static ReadOnlyMemory<byte> LatestIV { get; } = new byte[]
>     {
>         0x39, 0x6c, 0x34, 0x2c, 0x52, 0xa0, 0xc1, 0x2d,
>         0x51, 0x1d, 0xd0, 0x20, 0x9f, 0x90, 0xca, 0x7d,
>     };
> ```
this works:
![dc](https://user-images.githubusercontent.com/37000190/169633447-9b27bdfe-d4bc-44fc-8e88-f83694bfd1d7.png)

@alexrp here's a link for dc in case you need https://mega.nz/file/AZhzgCCb#04uWCyHrc73axrLNQFkMnOgOAFNGPCbmIvQTX-nCaog

**Reply by alexrp** [reply]
(Note that those warnings mean that the sheets in question will not be packed correctly back to a data center file until an XSD is provided, since attribute type information is taken from XSDs.)

**Comment by alexrp** [comment]
As of eb786b69100daa0ddb5ea7985cd4f5cd45f8b2fc, you can now pass `--encryption-key <hexstring>`/`--decryption-key <hexstring>` and `--encryption-iv <hexstring>`/`--decryption-iv <hexstring>` to various novadrop-dc commands.
